cmd/dcs-web/search/query-rewrite.go

```diff
@@ -2,11 +2,52 @@
 package search
 
 import (
-	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 )
 
+var (
+	start = regexp.MustCompile(`(?i)^\s*(-?(?:filetype|package|pkg|path|file)):(\S+)\s+`)
+	end = regexp.MustCompile(`(?i)\s+(-?(?:filetype|package|pkg|path|file)):(\S+)\s*$`)
+)
+
+func parseFilters(query *url.Values, qstr string, atstart bool) string {
+	q := *query
+
+	var re *regexp.Regexp
+	if atstart {
+		re = start
+	} else {
+		re = end
+	}
+
+	matches := re.FindStringSubmatchIndex(qstr)
+	for matches != nil {
+		// matches is [token_start, token_end, type_start, type_end, value_start, value_end]
```

Comment:
I was doing that so I have the indexes to slice off the matched part of the string.  I suppose I could use `re.FindStringSubmatch`, but then I'd have to use `strings.TrimPrefix`/`string.TrimSuffix`.  That seems like doing double work.